api/v1/type_cluster_config.go

```diff
@@ -8,21 +8,46 @@ import (
 )
 
 type UserFacingClusterConfig struct {
+	// Configuration options for the network feature.
 	Network       NetworkConfig       `json:"network,omitempty" yaml:"network,omitempty"`
+	// Configuration options for the dns feature.
 	DNS           DNSConfig           `json:"dns,omitempty" yaml:"dns,omitempty"`
+	// Configuration options for the ingress feature.
 	Ingress       IngressConfig       `json:"ingress,omitempty" yaml:"ingress,omitempty"`
+	// Configuration options for the load-balancer feature.
 	LoadBalancer  LoadBalancerConfig  `json:"load-balancer,omitempty" yaml:"load-balancer,omitempty"`
+	// Configuration options for the local-storage feature.
 	LocalStorage  LocalStorageConfig  `json:"local-storage,omitempty" yaml:"local-storage,omitempty"`
+	// Configuration options for the gateway feature.
 	Gateway       GatewayConfig       `json:"gateway,omitempty" yaml:"gateway,omitempty"`
+	// Configuration options for the metric server feature.
 	MetricsServer MetricsServerConfig `json:"metrics-server,omitempty" yaml:"metrics-server,omitempty"`
+	// Sets the cloud provider to be used by the cluster.
+	//
+	// When this is set as `external`, node will wait for an external cloud provider to
+	// do cloud specific setup and finish node initialization.
+	//
+	// Possible values: `external`.
 	CloudProvider *string             `json:"cloud-provider,omitempty" yaml:"cloud-provider,omitempty"`
 	Annotations   map[string]string   `json:"annotations,omitempty" yaml:"annotations,omitempty"`
```

Comment:
This should also be annotated and may contain a link to the possible anntoations in the docs, wdyt?

Reply:
Makes sense, although adding links to docstrings is a bit risky. We can't use relative paths since the docs are actually in a separate project and we can't link to https://documentation.ubuntu.com/canonical-kubernetes/latest/ either since the URLs may change, also it may refer to a different version.

I'd only say: ``Please refer to the ClusterAPI annotations reference for further details on these options.``

Reply:
Good point